### PR TITLE
 LTD-3123-filter-nca 

### PR DESCRIPTION
**Pull request by depsiatwal**

### CODE REVIEW

**Comment by depsiatwal** [comment]
This adds the filter to the UI to allow filter for cases that have a GoodOnApplication
is_nca_applicable

File: caseworker/cases/helpers/filters.py [hunk 1]
```diff
@@ -141,5 +141,12 @@ def case_filters_bar(request, filters, is_system_queue) -> FiltersBar:
                 title=CasesListPage.Filters.FLAGS,
                 options=[Option(flag["id"], flag["name"]) for flag in get_flags(request, disable_pagination=True)],
             ),
+            Checkboxes(
+                name="is_nca_applicable",
+                options=[
+                    Option(True, "Filter by Nuclear Cooperation Agreement", id="show-nuclear-cooperation-agreement")
+                ],
+                classes=["govuk-checkboxes--small"],
+            ),
```
Comment (currycoder) [inline caseworker/cases/helpers/filters.py:150]: Is there any way to test this?

Comment (Harriethw) [reply caseworker/cases/helpers/filters.py:150]: don't know if this is right but in this PR we couldn't find any existing tests (except UI ones that are currently skipped) so added a couple to at least check the box is checked/unchecked based on params https://github.com/uktrade/lite-frontend/pull/1018/commits/d6d46f84d3a2f817134b35da3b09f1ef17ad491e

Comment (depsiatwal) [reply caseworker/cases/helpers/filters.py:150]: added some tests 
we have some unit tests that do a get on 'cases' page and checks response. However I have added a couple of extra ones now as a baseline. One tests the context and the other sets a filter and checks the response of the api call.

**Pull request closed without merging**